File: src/main/java/ru/mm/surv/web/ThumbController.java
package ru.mm.surv.web;

import lombok.SneakyThrows;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.core.io.InputStreamResource;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import ru.mm.surv.capture.service.FfmpegStream;

import java.io.File;
import java.io.FileInputStream;
import java.io.InputStream;

@RestController
@RequestMapping("stream/thumb")
public class ThumbController {

    private final FfmpegStream ffmpegStream;

    @Autowired
    public ThumbController(FfmpegStream ffmpegStream) {
        this.ffmpegStream = ffmpegStream;
    }

    @GetMapping(value = "/{stream}/thumb.jpg", produces = MediaType.IMAGE_JPEG_VALUE)
    public ResponseEntity<InputStreamResource> thumb(@PathVariable("stream") String stream) {
        return ffmpegStream
                .getThumb(stream)
                .map(this::createInputStream)
                .map(InputStreamResource::new)
                .map(ResponseEntity::ok)
                .orElseGet(() -> ResponseEntity.notFound().build());
    }

    @SneakyThrows
    private InputStream createInputStream(File file) {
        return new FileInputStream(file);
    }
}

File: src/main/java/ru/mm/surv/web/ThumbController.kt
package ru.mm.surv.web

import org.springframework.web.bind.annotation.RestController
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.beans.factory.annotation.Autowired
import ru.mm.surv.capture.service.FfmpegStream
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.http.ResponseEntity
import org.springframework.core.io.InputStreamResource
import org.springframework.http.MediaType
import java.io.FileInputStream

@RestController
@RequestMapping("stream/thumb")
class ThumbController @Autowired constructor(private val ffmpegStream: FfmpegStream) {
    @GetMapping(value = ["/{stream}/thumb.jpg"], produces = [MediaType.IMAGE_JPEG_VALUE])
    fun thumb(@PathVariable("stream") stream: String): ResponseEntity<InputStreamResource> {
        return ffmpegStream
            .getThumb(stream)
            ?.let(::FileInputStream)
            ?.let(::InputStreamResource)
            ?.let { ResponseEntity.ok(it) }
            ?:ResponseEntity.notFound().build()
    }
}
